dlup/annotations.py

```diff
@@ -156,118 +327,158 @@ def __str__(self) -> str:
+
+    def intersect_with_box(self, other: ShapelyPolygon) -> Optional[list["Polygon"]]:
+        pre_area = self.area
+        if not self.is_valid:
+            valid_polygon = make_valid(self)
+        else:
+            valid_polygon = self
+
+        result = valid_polygon.intersection(other)
+        post_area = result.area
+        if pre_area > 0 and post_area == 0:
+            return None
+
+        annotation_class = self.annotation_class
+
+        if self.annotation_type == AnnotationType.BOX:
+            # Verify if this is still a box
+            if not _is_rectangle(shapely.geometry.mapping(result)["coordinates"]):
+                annotation_class = AnnotationClass(
+                    label=self.label,
+                    annotation_type=AnnotationType.POLYGON,
+                    color=self.color,
+                    z_index=self.z_index,
+                )
+
+        if isinstance(result, ShapelyPolygon):
+            return [Polygon(result, a_cls=annotation_class)]
+        elif isinstance(result, ShapelyMultiPolygon):
+            return [Polygon(geom, a_cls=annotation_class) for geom in result.geoms if geom.area > 0]
+        elif isinstance(result, shapely.geometry.collection.GeometryCollection):
+            return [Polygon(geom, a_cls=annotation_class) for geom in result.geoms if geom.area > 0]
+        else:
+            raise NotImplementedError(f"{type(result)}")
+
+    def __new__(cls, coords: Union[tuple[float, float], ShapelyPolygon], *args: Any, **kwargs: Any) -> "Polygon":
+        if isinstance(coords, ShapelyPolygon):
+            instance = super().__new__(cls, coords.exterior.coords, [ring.coords for ring in coords.interiors])
+        else:
+            instance = super().__new__(cls, coords)
+        instance.__class__ = cls
+        return cast("Polygon", instance)
 
     def __del__(self) -> None:
-        del self._id_to_attrs[str(id(self))]
+        if str(id(self)) in self._id_to_attrs:
+            del self._id_to_attrs[str(id(self))]
 
     def __getattr__(self, name: str) -> Any:
         try:
             return Polygon._id_to_attrs[str(id(self))][name]
         except KeyError as e:
-            raise AttributeError(str(e)) from None
+            raise AttributeError(f"{name} attribute not found") from e
 
     def __str__(self) -> str:
         return f"{self.annotation_class}, {self.wkt}"
 
 
-def rescale_geometry(geometry: Union[Point, Polygon], scaling: float | None = None) -> Union[Point, Polygon]:
-    if scaling is None:
-        return geometry
-    if scaling == 1.0:
-        return geometry
-
-    scaled_geometry = shapely.affinity.scale(geometry, scaling, scaling)
-    if isinstance(geometry, Polygon):
-        return Polygon(scaled_geometry, a_cls=geometry.annotation_class)
-
-    if isinstance(geometry, Point):
-        return Point(scaled_geometry, a_cls=geometry.annotation_class)
-
-    raise AnnotationError(f"geometry type {type(geometry)} is not a valid dlup type.")
-
-
```

Comment:
It can be useful, but I do not see a proper use case for this. I think you should always save the coordinates in level 0. The reason why it is there for ASAP is because they sometimes give them together with images resampled to 0.5um/pixel. If you want to read them in the native resolution you would need a scaling factor. 

It also behaved a bit erratic

Reply:
I was thinking of the use case when working with down-scaled data or feature vectors generated by models. For example, in [ahcore's _ValidationDataset](https://github.com/NKI-AI/ahcore/blob/a56f12326aec44b4c15bb1412813d6e89b310bfd/ahcore/utils/callbacks.py#L30) or with the ahcore's ZarrReader backend, the annotations are not always in level 0. Rescaling the geometry of `WsiAnnotations` might be necessary in these cases.

Reply:
Yes, but those are masks, so that’s not a problem there. If we save, we can always rescale when we save